acquia/acquia_cms
# ACMS-1289: Refactor role permissions.

**Motivation**
Refactor role permissions,

**Proposed changes**
Refactor Roles Permissions, move permission to specific modules.
Acquia CMS Common
- Removed permissions related to site studio, toolbar and tour from roles config file.
Acquia CMS Site Studio
- Added roles permissions related to site studio in hook_install and hook_update
Acquia CMS Toolbar
- Added roles permissions related to toolbar in hook_install and hook_update
Acquia CMS Tour
- Added roles permissions related to tour in hook_install and hook_update

**Alternatives considered**
PHP Unit test related to role permissions i.e. [BasicPermissionsTest](https://github.com/acquia/acquia_cms_common/blob/1.4.6/tests/src/Functional/BasicPermissionsTest.php) will fail.

**Testing steps**
New Site:
- Install site with this branch
- Roles should have the permissions listed in [BasicPermissionsTest](https://github.com/acquia/acquia_cms_common/blob/1.4.6/tests/src/Functional/BasicPermissionsTest.php)

Existing Site:
- Update codebase to this branch
- Run database update 
- - Roles should have the permissions listed in [BasicPermissionsTest](https://github.com/acquia/acquia_cms_common/blob/1.4.6/tests/src/Functional/BasicPermissionsTest.php)

- Other PHPUnit tests should pass.
- Manual Testing.

**Merge requirements**
